### PR TITLE
core: Serialize application task & stage

### PR DESCRIPTION
**Pull request by nwwebb**
Created a stage and task for serializing an application. Modeled it closely after the stages and tasks in orca. I tested it by sending a curl command to gate and it properly forwards the request to clouddriver. @lwander @duftler PTAL.

### CODE REVIEW

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/serialize/SerializeApplicationTask.groovy [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      .toBlocking()
+      .first()
+    Map outputs = [
+      "notification.type"  : "destroyjob",
+      "kato.last.task.id"  : taskId,
+      "delete.name"        : stage.context.jobName,
+      "delete.region"      : stage.context.region,
+      "delete.account.name": account
+    ]
+    new DefaultTaskResult(ExecutionStatus.SUCCEEDED, outputs)
```
Comment (lwander) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/serialize/SerializeApplicationTask.groovy]: Put a `return` here

**Comment by lwander** [comment]
LGTM

**Comment by lwander** [comment]
Needs a rebase

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/serialize/SerializeApplicationTask.groovy [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  @Override
+  TaskResult execute(Stage stage) {
+    String cloudProvider = getCloudProvider(stage)
+    String account = getCredentials(stage)
+    def taskId = kato.requestOperations(cloudProvider, [[serializeApplication: stage.context]])
+      .toBlocking()
+      .first()
+    Map outputs = [
+      "notification.type"  : "destroyjob",
```
Comment (duftler) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/serialize/SerializeApplicationTask.groovy]: This looks like a copy/paste error to me.

Comment (duftler) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/serialize/SerializeApplicationTask.groovy]: Actually, the three `delete.*` values below do as well.
The idea with the stage outputs is to populate it with contextual information that may be useful to downstream stages (or for script callers, or for troubleshooting, ...). So maybe the name of the application being serialized and related bits?

**Pull request merged**